### PR TITLE
feat: upgrade to purescript-argonaut-codecs with typed errors (JsonDecodeError instead of String)

### PR DESCRIPTION
**Pull request by srghma**
## What does this pull request do?

upgrades to https://github.com/purescript-contrib/purescript-argonaut-codecs/pull/73

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
   },
   "devDependencies": {
     "purescript-assert": "^4.1.0",
-    "purescript-console": "^4.2.0"
+    "purescript-console": "^4.2.0",
+    "purescript-exceptions": "^4.0.0"
```
Comment (thomashoneyman) [inline bower.json:27]: Since `exceptions` is used in the library code this would need to be under the `dependencies` key, not `devDependencies`.

Comment (thomashoneyman) [reply bower.json:27]: Whoops, I misread this — it looks like this is only used in the test code, so you were right putting that in dev dependencies in the first place.

**Comment by thomashoneyman** [comment]
Looks like CI is failing because the bower.json file is malformed.

**Comment by thomashoneyman** [comment]
Could you also move exceptions back into the dev dependencies?

**Comment by srghma** [comment]
yes, done

**Comment by thomashoneyman** [comment]
This still has CI failures. Did you update the argonaut-codecs version?

**Comment by srghma** [comment]
could you do that for me?)

**Pull request merged**